code/scpui/rocket_ui.cpp

```diff
@@ -543,13 +551,25 @@ void initialize()
 	os::events::addEventListener(SDL_KEYDOWN, os::events::DEFAULT_LISTENER_WEIGHT - 10, key_event_handler);
 
 	os::events::addEventListener(SDL_TEXTINPUT, os::events::DEFAULT_LISTENER_WEIGHT - 10, text_input_handler);
+
+	rocket_initialized = true;
 }
 void setOffset(float x, float y) { rendering_interface->setRenderOffset({x, y}); }
 
-void shutdown_scripting() { Rocket::Core::Lua::Interpreter::Shutdown(false); }
+void shutdown_scripting()
```

Comment:
Should `rocket_initialized = false;` be added in this function as well?

Reply:
No since that only disables the Lua part of libRocket and if `rocket_initialized` is set to `false` here then the actual `shutdown` call will return early without actually shutting down libRocket.

Reply:
Gotcha.